src/mechanisms/aes256cbc.rs

```diff
@@ -51,10 +50,8 @@ impl MechanismImpl for super::Aes256Cbc {
         // hprintln!(" aes256cbc encrypting l = {}B: {:?}", l, &buffer).ok();
 
         // Encrypt message in-place.
-        // &buffer[..pos] is used as a message and &buffer[pos..] as a reserved space for padding.
-        // The padding space should be big enough for padding, otherwise method will return Err(BlockModeError).
         let ciphertext = cipher
-            .encrypt_padded_mut::<ZeroPadding>(&mut buffer, l)
+            .encrypt_padded_mut::<NoPadding>(&mut buffer, l)
```

Comment:
Maybe don't panic if the padding is wrong, and error instead

Reply:
Right, fixed.